gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```diff
@@ -365,20 +369,51 @@ public Void call() throws Exception {
     return workUnitFile.toString();
   }
 
-  private void waitForJobCompletion() throws InterruptedException {
-    while (true) {
+  private void waitForJobCompletion()  throws InterruptedException {
+    LOGGER.info("Waiting for job to complete...");
+    long timeoutInSeconds;
+    timeoutInSeconds = Long.parseLong(this.jobProps.getProperty(ConfigurationKeys.HELIX_JOB_TIMEOUT_SECONDS,
```

Comment:
Yes, the requirement of this ticket is to handle stuck helix tasks. Assuming any task can stuck, we will be cancelling them after some timeout. jobs can set the timeout value, otherwise the default timeout value will be used, which is 3 hours - same as azkaban's default SLA time

Reply:
Hmmm.  I'm worried about having a naive SLA breaking existing uses.  Is there no way to have a heartbeat style check to ensure that helix tasks are not stuck?

Reply:
I will give it a thought and talk to the Helix team.